### ISSUE TITLE
ifort-2017.2.174-GCC-6.3.0-2.27.eb contains mpi

### ISSUE THREAD

**Issue by henkela** [easybuilders/easybuild-easyconfigs#4410]
I'm trying to install intel-2017.02.eb, which builds ifort-2017.2.174-GCC-6.3.0-2.27.eb. For me, impi-2017.2.174-iccifort-2017.2.174-GCC-6.3.0-2.27.eb fails to build because I don't have the file stated in the easyconfig but that's a different issue. 
I was looking at the built module ifort/2017.2.174-GCC-6.3.0-2.27 and found 
prepend-path	 PATH /cluster/easybuild/software/compiler/ifort/2017.2.174-GCC-6.3.0-2.27/compilers_and_libraries_2017.2.174/linux/mpi/intel64/bin

Why is that? Shouldn't this be in the module impi or iimpi, only? 

Best, 
Andreas

**Comment by boegel** [comment]
The `ifort` easyblock leverages the `icc` easyblock, which indeed also considers `linux/mpi/intel64/bin` as a candidate location for `$PATH` (it probably shouldn't... @bartoldeman?).

Usually, that location is not there though; did you change which installation files are used for `ifort`?

Here's what I have:

```
$ ls $EASYBUILD_PREFIX/software/ifort/2017.2.174-GCC-6.3.0-2.27/compilers_and_libraries_2017.2.174/linux
bin  compiler  documentation  man  mpirt  pkg_bin
```

**Comment by bartoldeman** [comment]
This is related to #3793

**Comment by bartoldeman** [comment]
Also hpcugent/easybuild-easyblocks#1032

**Comment by henkela** [comment]
@boegel,
I see the same + the folder mpi. 
But I'm confused anyway. Our intel cluster edition Tarball contains icc,ifort,mkl,mpi,itac,composer,vtunes,advisor,... and in easyconfig for mpi the source is something like l_mpi...tgz which i don't have. If I change it to the same source like icc or ifort it fails because "activation type has to be changed". Couldn't figure out yet whether it's in easyconfig or easyblock.

**Comment by boegel** [comment]
@henkela I think the problem is that you are using a different installation file for ifort, which indeed includes not only the compilers but also a whole bunch of other stuff. Usually you can use the 'full' tarball as a drop-in replacement since we specify the components to be installed, but that's clearly not working out as expected here...

Are you using the vanilla version of `ifort-2017.2.174-GCC-6.3.0-2.27.eb`, or did you make changes to it?

For `impi`, the installation mechanism was changed since a license file is no longer required for the latest versions, see https://github.com/hpcugent/easybuild-easyblocks/pull/1117 .

Which EasyBuild version are you using?

Can you specify a more complete error for your problem with `impi`?

**Comment by henkela** [comment]
@boegel no, it's not vanilla anymore. I had to uncomment the patches because that gave me an error. But I guess it's that patch that removes mpi... And that patches maybe only works on composer edition but we have cluster edition. Sorry, it's still difficult to see through easybuild structure... 

I'm running easybuild 3.1.1 

If I use vanilla  `impi-2017.2.174-iccifort-2017.2.174-GCC-6.3.0-2.27.eb` I get the error that easybuild doesn't find `l_mpi_%(version)s.tgz`. Which is ok, since I don't have that file. I exchange the source with my file and the checksum accordingly. If I run eb with that I get

`/cluster/easybuild/software/EasyBuild/3.1.1/lib/python2.7/site-packages/easybuild_easyconfigs-3.1.1-py2.7.egg/easybuild/easyconfigs/i/impi/impi-2017.2.174-iccifort-2017.2.174-GCC-6.3.0-2.27.eb failed (err: 'build failed (first 300 chars): cmd "./install.sh  -s /dev/shm/impi/2017.2.174/iccifort-2017.2.174-GCC-6.3.0-2.27/parallel_studio_xe_2017_update2/silent.cfg" exited with exitcode 2 and output:\nThe product you are installing does not support  activation type. Use other\nactivation options.\n-`

**Comment by boegel** [comment]
considering fixed in #3793, ok to close @henkela?

**Comment by Micket** [comment]
Long since outdated given we have moved into intel-compilers and oneapi sources (and this was probably also fixed back in 2017)
